hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieClusteringJob.java

```diff
@@ -121,17 +141,29 @@ public static void main(String[] args) {
   public int cluster(int retry) {
     this.fs = FSUtils.getFs(cfg.basePath, jsc.hadoopConfiguration());
     int ret = UtilHelpers.retry(retry, () -> {
-      if (cfg.runSchedule) {
-        LOG.info("Do schedule");
-        Option<String> instantTime = doSchedule(jsc);
-        int result = instantTime.isPresent() ? 0 : -1;
-        if (result == 0) {
-          LOG.info("The schedule instant time is " + instantTime.get());
+      String runningMode = cfg.runningMode == null ? "" : cfg.runningMode.toLowerCase();
```

Comment:
"" -> can  we use Optional to check null?  ""  is confused

Reply:
> When developers call `public int cluster(int retry)` internally like
> 
> https://github.com/apache/hudi/blob/5804ad8e32ae05758ebc5e47f5d4fb4db371ab52/hudi-utilities/src/test/java/org/apache/hudi/utilities/functional/TestHoodieDeltaStreamer.java#L1069
> 
> 
> They may not set running mode config, so we need check this value to avoid NLP.

Thought about it carefully, setting  "" here as default value is not good enough, maybe we need to validate RunningMode based on cfg.runSchedule when users let RunningMode be null.